src/hooks/tests/use-drag-and-drop-file.test.tsx

```diff
@@ -0,0 +1,71 @@
+// To run tests, execute `npm run test -- src/hooks/tests/use-drag-and-drop-file.test.tsx` from the root directory
+
+import { render, createEvent, fireEvent, renderHook, act } from '@testing-library/react';
+import { useDragAndDropFile } from '../use-drag-and-drop-file';
+
+const DragComponent = ( { onFileDrop }: { onFileDrop: () => void } ) => {
+	const { dropRef, isDraggingOver } = useDragAndDropFile< HTMLDivElement >( { onFileDrop } );
+	return (
+		<div data-testid="test-drop-zone" ref={ dropRef }>
+			{ isDraggingOver ? 'Dragging Over' : 'Not Dragging Over' }
+		</div>
+	);
+};
+
+describe( 'useDragAndDropFile', () => {
+	const onFileDrop = jest.fn();
+	beforeEach( () => {
+		onFileDrop.mockReset();
+	} );
+
+	test( 'should initialize with isDraggingOver as false', () => {
+		const { result } = renderHook( () => useDragAndDropFile( { onFileDrop } ) );
+		expect( result.current.isDraggingOver ).toBe( false );
+	} );
+	test( 'should set isDraggingOver to true on dragover event', async () => {
+		const { getByTestId, getByText } = render( <DragComponent onFileDrop={ onFileDrop } /> );
+		const dropZone = getByTestId( 'test-drop-zone' );
+
+		act( () => {
+			const dragOverEvent = createEvent.dragOver( dropZone );
+			fireEvent( dropZone, dragOverEvent );
+		} );
+
+		expect( getByText( 'Dragging Over' ) ).toBeInTheDocument();
+	} );
+
+	test( 'should set isDraggingOver to false on dragleave event', () => {
```

Comment:
I presume there's a type here:

```suggestion
	test( 'should set isDraggingOver to false on drag leave event', () => {
```

Reply:
Done https://github.com/Automattic/studio/pull/369/commits/d71d91da2273140f854bb79661db6324d7e1f913